e2e_test/iceberg/iceberg_sink_v2.slt

```diff
@@ -23,12 +23,14 @@ CREATE SINK s6 AS select * from mv6 WITH (
     connector = 'iceberg',
     type = 'append-only',
     force_append_only = 'true',
-    warehouse.path = 's3://icebergdata/demo',
-    s3.endpoint = 'http://127.0.0.1:9301',
-    s3.access.key = 'hummockadmin',
-    s3.secret.key = 'hummockadmin',
-    database.name='demo_db',
-    table.name='demo_table'
+    table.name = 'demo_db.demo_table',
+
+    iceberg.catalog.type = 'storage',
```

Comment:
Has this configuration format been stabilized in the iceberg community? It seems different to the iceberg configuration in other engine, such as spark. 

If it's not stabilized yet, I think we can keep the current configuration format, and manually fill in the map as the configuration of the current version of icelake. After the configuration is stabilized, we then introduce a breaking change PR to follow the official configuration.

Moreover, the current way that first assigns a name to catalog and then configure the catalog with its name is for multi-catalog usage. However, when we create a sink, it's unlikely to have multiple catalogs in a single statement. The current way is too verbose. cc @neverchanje

Reply:
This is the current configuration used by icelake, not stabilized in apache community yet. I don't think it's a good idea to keep current configuration since it's only designed for hadoop catalog. We will always need to break current syntax since we need to support rest catalog. 

> Moreover, the current way that first assigns a name to catalog and then configure the catalog with its name is for multi-catalog usage. 

All configurations starts with `iceberg.` are processed and recognized by icelake, not risingwave.

Reply:
Fixed